hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3726,6 +3731,16 @@ func (r *HostedClusterReconciler) validatePublishingStrategyMapping(hc *hyperv1.
 	return nil
 }
 
+// validateHostedClusterName validates a HostedCluster name confirms to RFC1123 convention
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+func (r *HostedClusterReconciler) validateHostedClusterName(hc *hyperv1.HostedCluster) error {
+	errs := validation.IsDNS1123Label(hc.Name)
+	if len(errs) > 0 {
+		return fmt.Errorf("hostedcluster name failed RFC1123 validation: %s", strings.Join(errs[:], " "))
```

Comment:
what's the purpose of the `[:]` ?

Reply:
`IsDNS1123Label` returns an array of strings and I just wanted to join them as a single string. It let's you convert the array to a slice and it's a shorthand for saying: start at index 0 and go to len(myArray).

https://go.dev/ref/spec#Slice_expressions
https://stackoverflow.com/a/28799151